search/firehose.go

```diff
@@ -50,7 +53,15 @@ func (s *Server) RunIndexer(ctx context.Context) error {
 	go s.bf.Start()
 
 	d := websocket.DefaultDialer
-	con, _, err := d.Dial(fmt.Sprintf("%s/xrpc/com.atproto.sync.subscribeRepos?cursor=%d", s.bgshost, cur), http.Header{})
+	u, err := url.Parse(s.bgshost)
+	if err != nil {
+		return fmt.Errorf("invalid bgshost URI: %w", err)
+	}
+	u.Path = "xrpc/com.atproto.sync.subscribeRepos"
+	if cur != 0 {
+		u.RawPath = fmt.Sprintf("cursor=%d", cur)
+	}
```

Comment:
A `cursor` of 0 asks for the farthest back event that the host is willing to provide. Maybe we want to default the value to `-1` as s sentinel so we can allow for an explicit 0 cursor?

Doesn't need to be blocking but just a thing to think about.

Reply:
We'll probably want to be able to configure this, the same way Kafka lets you default a consumer group to "start at current offset", "start at earliest available", "start at a configured time delta", etc.

For now it is in the "zero means empty/nil" situation. I'm going to leave as-is, which I think is a sane default with the backfill stuff now handling broader enumeration of backfill work.

Reply:
Just clarifying, this means if you provide no cursor we'll start from as far back as the BGS/PDS will allow us, not from live.

Reply:
Looking at this again, I guess I had a typo and it should be `RawQuery` not `RawPath`.

But if we don't provide a `cursor` HTTP query param at all, shouldn't the PDS (or BGS) give us the current stream, not start at the beginning? The intent of this snippet of code is to not set the parameter at all, instead of defaulting to `cursor=0` (which would scroll back to oldest available).

Reply:
Oh oops yeah you're right, I read that wrong